components/suggest/src/store.rs

```diff
@@ -442,10 +452,19 @@ impl<S> SuggestStoreInner<S> {
         })
     }
 
+    fn dismiss(&self, suggestion: Suggestion) -> Result<()> {
+        if let Some(key) = suggestion.dismissal_key() {
+            self.dbs()?
+                .writer
+                .write(|dao| dao.insert_dismissal(Some(suggestion.provider()), key))?;
+        }
```

Comment:
If you went with my suggestion above, then we'd need to delete the suggestion from the database here.  Your changes make that easy though, since there's now a `dismissal_key` column.